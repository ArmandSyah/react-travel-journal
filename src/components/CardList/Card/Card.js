import React from "react";  
import PropTypes from "prop-types";

import CardInfo from "./CardInfo/CardInfo";

export default function Card({imageUrl, ...props}){
    return (
        <div data-testid="card" className="card">
            <img className="card--image" src={imageUrl}/>
            <CardInfo {...props} />
        </div>
    )
}

CardInfo.propTypes = {
    imageUrl: PropTypes.string,
    location: PropTypes.string,
    title: PropTypes.string,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
    description: PropTypes.string,
    googleMapsUrl: PropTypes.string
}