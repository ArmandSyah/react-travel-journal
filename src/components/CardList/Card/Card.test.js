import React from "react";
import { render, cleanup } from "@testing-library/react";

import Card from './Card';

const props = {
    title: "Mount Fuji",
    location: "Japan",
    googleMapsUrl: "https://goo.gl/maps/1DGM5WrWnATgkSNB8",
    startDate: "12 Jan, 2021",
    endDate: "24 Jan, 2021",
    description: "Mount Fuji is the tallest mountain in Japan, standing at 3,776 meters (12,380 feet). Mount Fuji is the single most popular tourist site in Japan, for both Japanese and foreign tourists.",
    imageUrl: "https://source.unsplash.com/WLxQvbMyfas"
}

afterEach(cleanup);

it('renders properly', () => {
    const { getByText } = render(<Card {...props} />)

    expect(getByText(props.title)).toBeInTheDocument();
    expect(getByText(props.description)).toBeInTheDocument();
})

it('renders', async () => {
    const { asFragment } = render(<Card {...props} />)
    expect(asFragment()).toMatchSnapshot();
});