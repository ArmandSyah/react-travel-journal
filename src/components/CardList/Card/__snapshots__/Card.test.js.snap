// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="card"
  >
    <img
      class="card--image"
      src="https://source.unsplash.com/WLxQvbMyfas"
    />
    <div
      class="cardinfo"
    >
      <div
        class="cardinfo--location"
      >
        <i
          class="fa-solid fa-location-dot"
        />
        <span>
          JAPAN
        </span>
        <a
          href="https://goo.gl/maps/1DGM5WrWnATgkSNB8"
        >
          View On Google Maps
        </a>
      </div>
      <h1>
        Mount Fuji
      </h1>
      <div
        class="cardinfo--maintext"
      >
        <span
          class="cardinfo--maintext--date"
        >
          12 Jan, 2021 - 24 Jan, 2021
        </span>
        <span>
          Mount Fuji is the tallest mountain in Japan, standing at 3,776 meters (12,380 feet). Mount Fuji is the single most popular tourist site in Japan, for both Japanese and foreign tourists.
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
